src/SerializerBuilder.php

```diff
@@ -499,7 +499,10 @@ public function setMetadataCache(CacheInterface $cache): self
         return $this;
     }
 
-    public function build(): SerializerInterface
+    /**
+     * @return SerializerInterface&ArrayTransformerInterface
```

Comment:
is this syntax supported? as I know php does not have yet intersection types

Reply:
Yes it is, seen here:
- https://blog.jetbrains.com/phpstorm/2018/09/phpstorm-2018-3-eap-183-2635-12/
- https://phpstan.org/blog/union-types-vs-intersection-types
- https://github.com/squizlabs/PHP_CodeSniffer/pull/2919
- https://psalm.dev/docs/annotating_code/type_syntax/intersection_types/